File: ui-ios/Sources/ThreadwireUI/AttachmentPickerPopover.swift
import SwiftUI

/// Paperclip menu - "Photo"/"File" chrome only, both inert (M3/media hasn't started, so
/// there's nothing real to wire yet - matches this milestone's "visually complete,
/// functionally inert" treatment for out-of-scope capability, same as the mic button).
/// A `Menu` rather than `.popover` - guaranteed iOS 16.0-compatible without needing
/// `presentationCompactAdaptation` (iOS 16.4+, above this project's floor).
struct AttachmentPickerMenu: View {
    @Environment(\.threadwireColors) private var colors

    var body: some View {
        Menu {
            Button(action: {}) { Label("Photo", systemImage: "photo") }
            Button(action: {}) { Label("File", systemImage: "doc") }
        } label: {
            ThreadwireIconView(icon: .attach, color: colors.textSecondary, size: 20)
        }
        .accessibilityLabel("Add attachment")
    }
}

File: ui-android/src/main/kotlin/com/fsk/threadwire/ui/AttachmentPickerPopover.kt
package com.fsk.threadwire.ui

import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.automirrored.filled.InsertDriveFile
import androidx.compose.material.icons.filled.Photo
import androidx.compose.material3.DropdownMenu
import androidx.compose.material3.DropdownMenuItem
import androidx.compose.material3.Icon
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.graphics.vector.ImageVector
import com.fsk.threadwire.ui.theme.ThreadwireTheme

/**
 * Paperclip popover - "Photo"/"File" chrome only, both inert (M3/media hasn't started,
 * so there's nothing real to wire yet - matches this milestone's "visually complete,
 * functionally inert" treatment for out-of-scope capability, same as the mic button).
 */
@Composable
internal fun AttachmentPickerPopover(expanded: Boolean, onDismiss: () -> Unit) {
    DropdownMenu(expanded = expanded, onDismissRequest = onDismiss) {
        DropdownMenuItem(
            text = { Text("Photo", style = ThreadwireTheme.typography.msg) },
            leadingIcon = { PickerIcon(icon = Icons.Filled.Photo) },
            onClick = onDismiss,
        )
        DropdownMenuItem(
            text = { Text("File", style = ThreadwireTheme.typography.msg) },
            leadingIcon = { PickerIcon(icon = Icons.AutoMirrored.Filled.InsertDriveFile) },
            onClick = onDismiss,
        )
    }
}

@Composable
private fun PickerIcon(icon: ImageVector) {
    Icon(imageVector = icon, contentDescription = null, tint = ThreadwireTheme.colors.textSecondary)
}
